### PR TITLE
RAD2RAD compatibility with NLOCAL

### PR DESCRIPTION
**Pull request by bertrand-altair**
ENable compatibility betwenn NLOCAL method and rad2rad. NLOCAL dof are treated by rad2ad as classical dof, but new exchanges are needed.

### CODE REVIEW

File: engine/source/coupling/rad2rad/rad2rad_c.c [hunk 1]
```diff
@@ -675,6 +674,72 @@ int *igd, *nng, *nodbuf, *itab,*addcnel,*cnel,*ixc,*ofc,*info,*typ,*cdt,*cdr,*pr
...
+my_real_c *crd;
+
+    /************************coupling for non local dof***************************/
+    /*****************************************************************************/
+    lbuf = *nb_tot_dof*sizeof(my_real_c);
+    crd  = (my_real_c *) malloc(3*lbuf);
+    lbuf1  = *nb_tot_dof*sizeof(int);   
+    nodid  = (int *) malloc(lbuf1);
+    bcs  = (int *) malloc(lbuf1);
+    printf("lbuf %d %d %d\n",*nb_tot_dof,lbuf,lbuf1);
```
Comment (sebastienVilleneuve) [inline engine/source/coupling/rad2rad/rad2rad_c.c:692]: @bertrand-altair 

here there is a print that you certainly missed to remove

File: engine/source/coupling/rad2rad/rad2rad_c.c [hunk 2]
```diff
@@ -675,6 +674,72 @@ int *igd, *nng, *nodbuf, *itab,*addcnel,*cnel,*ixc,*ofc,*info,*typ,*cdt,*cdr,*pr
...
+int *nodid,*bcs;
+my_real_c *crd;
+
+    /************************coupling for non local dof***************************/
+    /*****************************************************************************/
+    lbuf = *nb_tot_dof*sizeof(my_real_c);
+    crd  = (my_real_c *) malloc(3*lbuf);
+    lbuf1  = *nb_tot_dof*sizeof(int);   
+    nodid  = (int *) malloc(lbuf1);
+    bcs  = (int *) malloc(lbuf1);
```
Comment (sebastienVilleneuve) [inline engine/source/coupling/rad2rad/rad2rad_c.c:691]: @bertrand-altair 

I think you need to free the memory at the end of routine

File: engine/source/coupling/rad2rad/rad2rad_c.c [hunk 3]
```diff
@@ -1123,7 +1188,45 @@ int *idp, *nng, *nodbuf;
...
+    lbuf = *nng*sizeof(my_real_c);
+    mbuf = (my_real_c *) malloc(lbuf);
+
+    for (i = 0; i < *nng; i++)
+      {
+       mbuf[i] = ms[iadd_nl[i]-1];
+      }
+    
+    writer(fidw, (void *) mbuf, lbuf);    
+    free(mbuf);
```
Comment (sebastienVilleneuve) [inline engine/source/coupling/rad2rad/rad2rad_c.c:1211]: @bertrand-altair 

I think free of lbuf is missing

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
@bertrand-altair 

I've saw some modifications to perform

Thanks,
Sébastien

File: engine/source/coupling/rad2rad/rad2rad_c.c (hunk 3 above)
Comment (bertrand-altair) [reply engine/source/coupling/rad2rad/rad2rad_c.c:1211]: lbuf is not an array

**Pull request closed without merging**